File: tags/symmetric-2.0.8/src/main/java/org/jumpmind/symmetric/model/DataMetaData.java
/*
 * SymmetricDS is an open source database synchronization solution.
 *   
 * Copyright (C) Chris Henson <chenson42@users.sourceforge.net>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, see
 * <http://www.gnu.org/licenses/>.
 */
package org.jumpmind.symmetric.model;

import org.apache.ddlutils.model.Table;

public class DataMetaData {

    private Data data;
    private Table table;
    private TriggerRouter triggerRouter;
    private NodeChannel nodeChannel;

    public DataMetaData(Data data, Table table, TriggerRouter trigger, NodeChannel nodeChannel) {
        this.data = data;
        this.table = table;
        this.triggerRouter = trigger;
        this.nodeChannel = nodeChannel;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }

    public Table getTable() {
        return table;
    }

    public void setTable(Table table) {
        this.table = table;
    }

    public TriggerRouter getTriggerRouter() {
        return triggerRouter;
    }

    public void setTriggerRouter(TriggerRouter trigger) {
        this.triggerRouter = trigger;
    }

    public NodeChannel getNodeChannel() {
        return nodeChannel;
    }

    public void setNodeChannel(NodeChannel nodeChannel) {
        this.nodeChannel = nodeChannel;
    }

    public TriggerHistory getTriggerHistory() {
        return data != null ? data.getTriggerHistory() : null;
    }

}

File: symmetric-client-clib/src/model/DataMetaData.c
/**
 * Licensed to JumpMind Inc under one or more contributor
 * license agreements.  See the NOTICE file distributed
 * with this work for additional information regarding
 * copyright ownership.  JumpMind Inc licenses this file
 * to you under the GNU General Public License, version 3.0 (GPLv3)
 * (the "License"); you may not use this file except in compliance
 * with the License.
 *
 * You should have received a copy of the GNU General Public License,
 * version 3.0 (GPLv3) along with this library; if not, see
 * <http://www.gnu.org/licenses/>.
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
#include "model/DataMetaData.h"

void SymDataMetaData_destroy(SymDataMetaData *this) {
    free(this);
}

SymDataMetaData * SymDataMetaData_new(SymDataMetaData *this, SymData *data, SymTable *table, SymRouter *router, SymChannel *nodeChannel) {
    if (this == NULL) {
        this = (SymDataMetaData *) calloc(1, sizeof(SymDataMetaData));
    }
    this->data = data;
    this->table = table;
    this->router = router;
    this->nodeChannel = nodeChannel;
    this->destroy = (void *) SymDataMetaData_destroy;
    return this;
}
